### PR TITLE
[WIP]SyftCrypTensor is created

### PR DESCRIPTION
**Pull request by ajnovice**
Functionality is not completed yet. It's a work in progress [WIP]

### CODE REVIEW

File: test/torch/tensors/test_crypten_tensor.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    ptype_values = [Ptype.arithmetic, Ptype.binary]
+    tensor_types = [ArithmeticSharedTensor, BinarySharedTensor]
+    for i, curr_ptype in enumerate(ptype_values):
+        tensor = MPCTensor(x, ptype=curr_ptype)
+        syft_crypten = SyftCrypTensor(tensor=tensor)
+        assert isinstance(syft_crypten.tensor._tensor, tensor_types[i])
+
+
+def test_to():
+    # TODO need to check the internal structute if it is correctly changed or not!
```
Comment (gmuraru) [inline test/torch/tensors/test_crypten_tensor.py:36]: TYPO: ```structute``` -> ```structure```

File: syft/frameworks/torch/tensors/crypten/syft_crypten.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+        self.tensor = self.tensor.to(ptype, **kwargs)
+        return self
+
+    def arithmetic(self):
+        """Converts self._tensor to arithmetic secret sharing"""
+        self.tensor = self.tensor.arithmetic()
+        return self
+
+    def binary(self):
+        """Converts self._tensor to binary secret sharing"""
```
Comment (gmuraru) [inline syft/frameworks/torch/tensors/crypten/syft_crypten.py:51]: Typo: ```_tensor``` --> ```tensor```

File: test/torch/tensors/test_crypten_tensor.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+import crypten
+import torch
+from crypten.mpc import MPCTensor
+from crypten.mpc.primitives import ArithmeticSharedTensor, BinarySharedTensor
+from crypten.mpc.ptype import ptype as Ptype
+
+from syft.frameworks.torch.tensors.crypten.syft_crypten import SyftCrypTensor
+
+crypten.init()
```
Comment (gmuraru) [inline test/torch/tensors/test_crypten_tensor.py:9]: Q: Maybe is better to have a ```setup``` method and a ```destroy``` one where we call ```init``` and ```uninit```?

File: test/torch/tensors/test_crypten_tensor.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def test_repr():
+    x = torch.tensor([0.1, 0.2, 0.3])
+    arithmetic = MPCTensor(x, ptype=Ptype.arithmetic)
+    binary = MPCTensor(x, ptype=Ptype.binary)
+    arithmetic_syft_crypten = SyftCrypTensor(tensor=arithmetic)
+    binary_syft_crypten = SyftCrypTensor(tensor=binary)
+    # Make sure these don't crash
+    print(arithmetic_syft_crypten)
```
Comment (gmuraru) [inline test/torch/tensors/test_crypten_tensor.py:19]: Q: Shouldn't we test the representation with an ```assert```?

File: syft/__init__.py [hunk 5]
```diff
@@ -113,6 +113,7 @@ def pool():
         "AutogradTensor",
         "FixedPrecisionTensor",
         "LargePrecisionTensor",
+        "SyftCrypTensor",
```
Comment (youben11) [inline syft/__init__.py:116]: I guess SyftCrypTensor shouldn't be made available to be imported from syft

File: test/torch/tensors/test_crypten_tensor.py [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+
+
+def test_type():
+    x = torch.tensor([0.1, 0.2, 0.3])
+    ptype_values = [Ptype.arithmetic, Ptype.binary]
+    tensor_types = [ArithmeticSharedTensor, BinarySharedTensor]
+    for i, curr_ptype in enumerate(ptype_values):
+        tensor = MPCTensor(x, ptype=curr_ptype)
+        syft_crypten = SyftCrypTensor(tensor=tensor)
+        assert isinstance(syft_crypten.tensor._tensor, tensor_types[i])
```
Comment (gmuraru) [inline test/torch/tensors/test_crypten_tensor.py:32]: Q: Can't we simply test the ```tensor```?

File: test/torch/tensors/test_crypten_tensor.py [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+def test_to():
+    # TODO need to check the internal structute if it is correctly changed or not!
+    x = torch.tensor([0.1, 0.2, 0.3])
+    ptype_values = [Ptype.arithmetic, Ptype.binary]
+    tensor_types = [ArithmeticSharedTensor, BinarySharedTensor]
+    for i, curr_ptype in enumerate(ptype_values):
+        mpc = MPCTensor(x)
+        tensor = SyftCrypTensor(tensor=mpc)
+        syft_crypten1 = tensor.to(ptype_values[i])
```
Comment (gmuraru) [inline test/torch/tensors/test_crypten_tensor.py:43]: MAYBE: Name it simply ```syft_crypten```

**Pull request closed without merging**